### DISCUSSION TITLE
[ISSUE] Unable to create a cart using storefront api for a bundle product

### DISCUSSION THREAD

**Discussion by vamshiaruru** [Shopify/storefront-api-feedback#229] [Help]
Hi everyone, we are a shopify plus store using bundles, and we are using storefront api to build our UI. However, it doesn't seem like we can add a bundle product to cart when using storefront api. I think it is because the bundle product is available only on online store channel, and not published on our app channel?
What do we do here? This is a major issue for us because we will not be able to support bundles at all in our store. Any help is very appreciated.

**Comment by kmdavis** [comment]
Hi! Bundles ATC here. In order to debug this further, we need a bit more information:
1. what is your shop domain and/or shop id?
2. what is the id of the channel you are using?

**Comment by moritzlaube** [comment]
Hi! Has this been resolved? I am facing the same issue. I created a bundle of some of my products using Shopify's Bundles app. This creates a new product in the Shopify Dashboard. Querying the bundle's data also wirks just fine. But when trying to add the Bundle to the cart using the Storefront API, I get back an empty object inside the lineItems array while the cartTotal is updated correctly.

**Comment by AlinaSuhorebraya** [comment]
I checked my request and found out that for `merchandiseId` I had "gid://shopify/Product/{id}" but it should be "gid://shopify/ProductVariant/{id}" with id from variant field, so it helps me to fix it
